algobotdash/storage.py

```
@@ -267,42 +291,105 @@ def _normalize_timestamps(
         "CASE WHEN status = 'open' THEN NULL ELSE pnl END AS realized_pnl "
-        "FROM positions "
+        f"FROM positions {where_sql} "
         f"ORDER BY {column} {order}, position_id ASC LIMIT ? OFFSET ?"
     )
     connection = _open_projection(path)
     try:
         rows, total = _page_rows(
             connection,
             query,
-            "SELECT COUNT(*) FROM positions",
+            f"SELECT COUNT(*) FROM positions {where_sql}",
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)